api-registration/server/index.js

```diff
@@ -0,0 +1,78 @@
+require('dotenv/config');
+const pg = require('pg');
+const argon2 = require('argon2'); // eslint-disable-line
+const express = require('express');
+const ClientError = require('./client-error');
+const errorMiddleware = require('./error-middleware');
+
+const db = new pg.Pool({ // eslint-disable-line
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+
+app.post('/api/auth/sign-up', (req, res, next) => {
+  const { username, password } = req.body;
+  if (!username || !password) {
+    throw new ClientError(400, 'username and password are required fields');
+  }
+
+  /* your code starts here */
+  argon2
+    .hash(password)
+    .then(hashedPassword => {
+      const sql = `
+        insert into "users" ("username", "hashedPassword")
+        values ($1, $2)
+        returning*;
+        `;
+      const params = [username, hashedPassword];
+      db.query(sql, params)
+        .then(result => {
+          const newRow = result.rows[0];
+          const returnObject = {};
+          returnObject.createdAt = newRow.createdAt;
+          returnObject.userId = newRow.userId;
+          returnObject.username = newRow.username;
+          res.status(201).json(returnObject);
```

Comment:
This works and is fine but as an FYI you can simplify this by setting which columns are returned from the SQL statement.
```JAVASCRIPT
const sql = `
  insert into "users" ("username", "hashedPassword")
  values ($1, $2)
  returning "createdAt", "userId", "username";
`;
const params = [username, hashedPassword];
db.query(sql, params)
  .then(result => {
    const newUser = result.rows[0];
    res.status(201).json(returnUser);
```